test/utils/ReactDOMTestContainer.js

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import {Simulate} from 'react-dom/test-utils';
+import {reactUniDriver} from 'unidriver';
+
+export function createDOMContainer() {
+  const root = document.querySelector('#root') || document.body;
+  const div = document.createElement('div');
+  root.appendChild(div);
+  return div;
+}
+
+// A wrapper around createDOMContainer to reduce boilerplate when working with
+// React components.
+export class ReactDOMTestContainer {
+  node = null;
+
+  get componentNode() {
+    return this.node.firstElementChild;
+  }
+
+  // The container is usually created outside of before() test hook, and the
+  // constructor runs before any of the tests, globally. Instead of attaching
+  // the DOM element upfront in the constructor and polluting the document we
+  // provide this method.
+  create() {
+    this.node = createDOMContainer();
+    return this;
+  }
+
+  destroy() {
+    this.node.remove();
+    return this;
+  }
+
+  renderAsync(jsx) {
+    return new Promise(resolve => resolve(ReactDOM.render(jsx, this.node)));
```

Comment:
here's the implementation of `ReactDOM.render` [here](https://github.com/facebook/react/blob/master/packages/react-dom/src/client/ReactDOM.js#L665-L668)

as you see it accepts `callback` as third argument. you can use that in order to make `renderAsync` resolve at correct time
```suggestion
    return new Promise(resolve => ReactDOM.render(jsx, this.node, resolve));
```

Reply:
Oh, I see